server/cluster.go

```diff
@@ -78,18 +79,40 @@ func newRaftCluster(s *Server, clusterID uint64) *RaftCluster {
 }
 
 func (c *RaftCluster) loadClusterStatus() (*ClusterStatus, error) {
-	data, err := c.s.kv.Load((c.s.kv.ClusterStatePath("raft_bootstrap_time")))
+	bootstrapTime, err := c.loadBootstrapTime()
 	if err != nil {
 		return nil, err
 	}
-	if len(data) == 0 {
-		return &ClusterStatus{}, nil
+	var isInitialized bool
+	if bootstrapTime != zeroTime {
+		isInitialized = c.isInitialized()
+	}
+	return &ClusterStatus{
+		RaftBootstrapTime: bootstrapTime,
+		IsInitialized:     isInitialized,
+	}, nil
+}
+
+func (c *RaftCluster) isInitialized() bool {
+	if c.cachedCluster.getRegionCount() > 1 {
+		return true
 	}
-	t, err := parseTimestamp([]byte(data))
+	region := c.cachedCluster.searchRegion(nil)
+	return region != nil &&
+		len(region.GetVoters()) >= int(c.s.GetReplicationConfig().MaxReplicas) &&
+		len(region.GetPendingPeers()) == 0
+}
+
+func (c *RaftCluster) loadBootstrapTime() (t time.Time, err error) {
+	var data string
+	data, err = c.s.kv.Load(c.s.kv.ClusterStatePath("raft_bootstrap_time"))
 	if err != nil {
-		return nil, err
+		return
```

Comment:
no because there are named return parameters. I avoid using named return parameters unless its clear that it simplifies the code. In this case I don't think it helps and it even forces a separate variable declaration line for `data`.